scripts/setup/bootstrap.sh

```diff
@@ -56,6 +56,29 @@ _install_additional_pip_requirements() {
     unset _SETUP_PLATFORM
 }
 
+_submodules_need_updating() {
+  # Validates if a set o submodules that should always be checked out are up to date
+
+  # Pigweed will be up to date on an initial setup, however it may change over time
+  # the rest are a small subset of things that are always checked out (have no platform attachment)
+  _SUBMODULE_PATHS=(
+    "third_party/pigweed/repo"
+    "third_party/openthread/repo"
+    "third_party/editline/repo"
+  )
+
+  for path in "${_SUBMODULE_PATHS[@]}"; do
+    if git submodule status $path | grep -E '^-' >/dev/null 2>&1; then 
```

Comment:
Updated.

Regarding shellharden, we explicitly ignore bootstrap: https://github.com/project-chip/connectedhomeip/blob/master/.restyled.yaml#L85 apparently because it wrongly tries to quote some things we do not want quoting.

Reply:
> because it wrongly tries to quote some things we do not want quoting

Hmm... OK, I'll try to look at it, maybe it's not the fault of shellcheck but the looping pattern itself...